src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,193 @@
+            }
+            String[] split = cmdSplit[1].split(" /at ");
+            c = new AddCommand("event", split[0], parseDate(split[1]));
+            break;
+        case "list":
+            c = new ListCommand();
+            break;
+        case "bye":
+            c = new ExitCommand();
+            break;
+        case "delete":
+            c = new DeleteCommand(parseNum(cmdSplit[1]));
+            break;
+        case "mark":
+            c = new UpdateCommand("mark", parseNum(cmdSplit[1]));
+            break;
+        case "unmark":
+            c = new UpdateCommand("unmark", parseNum(cmdSplit[1]));
+            break;
+        case "find":
+            c = new FindCommand(cmdSplit[1].toLowerCase());
+            break;
+        default:
+            throw new LunaInvalidCommandException();
+        }
+        return c;
+    }
+
+    /**
+     * Processes lines of text from the hard disk by converting strings of tasks into
+     * appropriate Task instances.
+     *
+     * @param tasks String of task to be converted to a Task instance
+     * @return A Task instance.
+     */
+    public static Task parseSaved(String tasks) {
+        String txt = tasks.substring(7);
+        if (txt.startsWith("[T]")) {
+            // Abstract task description
+            String[] split = txt.split("] ");
+
+            // Create new Todo
+            Task tsk = new Todo(split[1]);
+
+            // Update status
+            if (split[0].substring(3).equals("[✧")) {
+                tsk.setStatusIcon(true);
+            }
+            return tsk;
+        } else if (txt.startsWith("[D]")) {
+            // Abstract task description and date
+            String[] split = txt.split("] ");
+            String [] desSplit = split[1].split(" BY ");
+            String des = desSplit[0];
+            String by = desSplit[1].substring(0, 11);
+
+            // Create date
+            LocalDate date = LocalDate.parse(by, DateTimeFormatter.ofPattern("dd-MMM-yyyy"));
+
+            // Create new Deadline
+            Task tsk = new Deadline(des, date);
+
+            // Update status
+            if (split[0].substring(3).equals("[✧")) {
+                tsk.setStatusIcon(true);
+            }
+            return tsk;
+        } else if (txt.startsWith("[E]")) {
+            // Abstract task description and date
+            String[] split = txt.split("] ");
+            String [] desSplit = split[1].split(" AT ");
+            String des = desSplit[0];
+            String at = desSplit[1].substring(0, 11);
+
+            // Create date
+            LocalDate date = LocalDate.parse(at, DateTimeFormatter.ofPattern("dd-MMM-yyyy"));
+
+            // Create new Event
+            Task tsk = new Event(des, date);
+
+            // Update status
+            if (split[0].substring(3).equals("[✧")) {
+                tsk.setStatusIcon(true);
+            }
+            return tsk;
+        } else {
+            return null;
+        }
+    }
```

Comment:
I like your simple javadocs, and the comments to breakdown the process for readers